### PR TITLE
fix(aria-roles): Add WAI-ARIA 1.2 roles

### PR DESCRIPTION
**Pull request by timmybytes**
Adds new roles from WAI-ARIA 1.2 to lib/standards/aria-roles.js

Closes issue: #2107 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security by @WilcoFiers

### CODE REVIEW

**Comment by timmybytes** [comment]
Hey there. This is my first time working with a codebase of this size and complexity. I followed the contributing guidelines correctly (I think), only altered the single `aria-roles.js` file, but run into the following error:
```bash
  1 failing

  1) standards.getAriaRolesByType
       should return a list of role names by type:

      AssertionError: expected [ Array(37) ] to deeply equal [ Array(25) ]
      + expected - actual

       [
         "article"
      -  "blockquote"
      -  "caption"
         "cell"
      -  "code"
         "columnheader"
         "definition"
      -  "deletion"
         "directory"
         "document"
      -  "emphasis"
         "feed"
         "figure"
         "group"
         "heading"
         "img"
      -  "insertion"
         "list"
         "listitem"
         "math"
      -  "meter"
         "none"
         "note"
      -  "paragraph"
         "presentation"
         "row"
         "rowgroup"
         "rowheader"
         "separator"
      -  "strong"
      -  "subscript"
      -  "superscript"
         "table"
         "term"
      -  "time"
         "toolbar"
         "tooltip"
       ]
      
      at Context.<anonymous> (standards/get-aria-roles-by-type.js:15:10)
```

I've gone through the get-aria-roles-by-type.js file, which imports from lib/standards/index.js, but I'm at a bit of a loss as how to proceed. It seems like these new roles aren't being added in somewhere else for comparison checks, but like I said I'm a bit green with a project this complex. Could you point me in the right direction to get this fixed for you all?

**Comment by straker** [comment]
Sure. The [test that is failing])https://github.com/dequelabs/axe-core/blob/develop/test/commons/standards/get-aria-roles-by-type.js#L15-L41) takes the roles directly from the standards object. So the fix should just be to add all the roles (looks like you added them all as structure roles) to the list from that test and it should work.

File: lib/standards/aria-roles.js [hunk 1]
```diff
@@ -284,6 +303,10 @@ const ariaRoles = {
 		],
 		nameFromContent: true
 	},
+	meter: {
+		type: 'structure',
+		requiredAttrs: ['aria-valuemax', 'aria-valuemin', 'aria-valuenow']
```
Comment (straker) [inline lib/standards/aria-roles.js:308]: We'll need to add `aria-valuetext` in the `allowedAttrs` since we don't have inherited attrs.

**Review by straker — CHANGES_REQUESTED** [review]
Awesome work. Just need to add 1 attribute and should be good to go after the tests are fixed.

File: lib/standards/aria-roles.js (hunk 1 above)
Comment (timmybytes) [reply lib/standards/aria-roles.js:309]: Okay, I think I got it fixed.

**Review by straker — COMMENTED** [review]
LGTM. But I want @WilcoFiers to look it over really quick make sure this is what he intended.

**Comment by timmybytes** [comment]
Absolutely, no worries. Thanks so much for the help!

File: lib/standards/aria-roles.js [hunk 2]
```diff
@@ -39,11 +39,18 @@ const ariaRoles = {
...
+		type: 'structure'
+	},
 	button: {
 		type: 'widget',
 		allowedAttrs: ['aria-expanded', 'aria-pressed'],
 		nameFromContent: true
 	},
+	caption: {
+		type: 'structure',
+		requiredContext: ['figure', 'table', 'grid']
```
Comment (WilcoFiers) [inline lib/standards/aria-roles.js:52]:
```suggestion
		requiredContext: ['figure', 'table', 'grid', 'treegrid']
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

**Review by WilcoFiers — APPROVED** [review]

**Comment by WilcoFiers** [comment]
@timmybytes Thank you for the pull request! Much appreciated.

**Pull request merged**